src/scheduler/buckets.cpp

```diff
@@ -1134,11 +1134,11 @@ check_node_buckets(status *policy, server_info *sinfo, queue_info *qinfo, resour
 			move_schd_error(err, failerr);
 	} else if (!(sinfo->svr_to_psets.empty())) {
 		/* Find buckets associated with nodes of the server which owns the job */
-		for (auto spset = sinfo->svr_to_psets.begin(); spset != sinfo->svr_to_psets.end(); spset++) {
-			if (strcmp(spset->svr_inst_id, resresv->job->svr_inst_id) == 0) {
+		for (auto spset: sinfo->svr_to_psets) {
```

Comment:
this makes a copy of each pset in `sinfo->svr_to_psets`.
If you don't want to do this, you can iterate over references
`for (auto &spset : sinfo->svr_to_psets) {`

See [StackOverflow](https://stackoverflow.com/questions/15176104/c11-range-based-loop-get-item-by-value-or-reference-to-const) for a better explanation

Reply:
thanks for educating! will change it to use references